library/core/src/sync/atomic.rs

```diff
@@ -904,8 +904,22 @@ impl AtomicBool {
     #[cfg(target_has_atomic = "8")]
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     pub fn fetch_and(&self, val: bool, order: Ordering) -> bool {
-        // SAFETY: data races are prevented by atomic intrinsics.
-        unsafe { atomic_and(self.v.get(), val as u8, order) != 0 }
+        if cfg!(any(target_arch = "x86_64", target_arch = "x86")) {
```

Comment:
Are there other architectures that also generate a needless CAS loop? Might be worth leaving a comment with all the arches where this _doesn't_ matter, so people in the future know that their arch wasn't forgotten.

Reply:
There are some, I added list of those that doesn't need this.